src/scripts/mdSelect.js

```diff
@@ -77,8 +77,23 @@ function mdSelect($compile, $parse) {
       if(self.disabled) {
         return;
       }
+      
+      var indexOfItem = tableCtrl.selected.indexOf(self.model);
+
+      if(indexOfItem === -1 && self.id && tableCtrl.selected && tableCtrl.selected[0] !== undefined){
+        for(var key in tableCtrl.selected){
+          if(tableCtrl.selected[key].id === self.id){
+            indexOfItem = key;
+            break;
+          }
+        }
+      }
 
-      tableCtrl.selected.splice(tableCtrl.selected.indexOf(self.model), 1);
```

Comment:
Ther we have -1 after updating list becouse ref of object other. But id is same. We can find appropriate item by id.